train.py

```diff
@@ -73,12 +74,20 @@ def generate_subspectrogram(duration_s=config.SUBSPECTROGRAM_DURATION_S, fft_rat
         checkpoint.epoch.assign_add(1)
         checkpoint_manager.save()
 
-    print("======================== evaluation on validation data =========================")
-    # test model on validation set
-    utils.test_model(model, val_spectrograms, val_labels, val_loss, val_accuracy)
+        if epoch % 10 == 0 or epoch == config.NB_EPOCHS - 1:
+            print("======================== evaluation on validation data =========================")
+            # test model on validation set
+            val_accuracies.append(utils.test_model(
+                model, val_spectrograms, val_labels, val_loss, val_accuracy, epoch=epoch+1))
 
     utils.save_model(model, epoch)
-    utils.save_and_display_loss_through_epochs(epoch_range, loss_history, model.name)
+    utils.save_and_display_loss_through_epochs(epoch_range, loss_history, model.name, "training loss")
+
+    # Plot val accuracies
+    val_accuracy_range = [i for i in epoch_range if i % 10 == 0 or i == config.NB_EPOCHS - 1]
+    utils.save_and_display_loss_through_epochs(val_accuracy_range, val_accuracies, model.name, "validation accuracy")
+    best_epoch = (len(val_accuracies) - list(reversed(val_accuracies)).index(min(val_accuracies)) - 1) * 10
```

Comment:
Considering how jumpy the accuracy can be, I think we are better off checking the curve to be sure of the minima. We might want to plot accuracy as well

Reply:
Sorry that is actually the loss, forgot to change the variable name. 
The best epoch is just an indication, validation loss is plotted (l. 88). We could do both, I’ll add that.